standalone/pruntime/enclave/src/contracts/btc_lottery.rs

```diff
@@ -0,0 +1,359 @@
+                    Some(_) => self.lottery_set.get(&self.round_id).unwrap().clone(),
+                    None => BTreeMap::<String, PrivateKey>::new(),
+                };
+                for winner_id in sample {
+                    let mut salt = self.round_id * 10000;
+                    let s = Secp256k1::new();
+                    let raw_data = (raw_seed, salt);
+                    let seed = blake2_128(&Encode::encode(&raw_data));
+                    let pk = ExtendedPrivKey::new_master(Network::Bitcoin, &seed)
+                        .unwrap()
+                        .private_key;
+                    lottery_token.insert(String::from(winner_id), pk);
+                    salt += 1;
+                }
+                self.lottery_set.insert(self.round_id, lottery_token);
+                self.round_id += 1;
+            } else if let chain::pallet_kitties::RawEvent::Open(round_id, token_id, btc_address) =
+                pe
+            {
+                let token_id = format!("{:#x}", token_id);
+                let btc_address = format!("{:#x}", btc_address);
+                let sender = AccountIdWrapper::from_hex(ALICE);
+                let target = Address::from_str(&btc_address).unwrap();
+                let sequence = self.sequence + 1;
+                let data;
+                if (!self
+                    .lottery_set
+                    .get(&round_id)
+                    .unwrap()
+                    .contains_key(&token_id))
+                {
+                    data = BtcTransfer {
+                        dest: sender.clone(),
+                        tx: Vec::new(),
+                        sequence,
+                    };
+                } else {
+                    let secp = Secp256k1::new();
+                    let private_key: PrivateKey = *self
+                        .lottery_set
+                        .get(&round_id)
+                        .unwrap()
+                        .get(&token_id)
+                        .unwrap();
+                    let public_key = PublicKey::from_private_key(&secp, &private_key);
+                    let private_addr = Address::p2pkh(&public_key, Network::Bitcoin);
+                    let round_utxo = self.utxo.get(&round_id).unwrap();
+                    let (txid, vout, amount) = round_utxo.get(&private_addr).unwrap();
+                    let mut tx = Transaction {
+                        input: vec![TxIn {
+                            previous_output: OutPoint {
+                                txid: *txid,
+                                vout: *vout,
+                            },
+                            sequence: RBF,
+                            witness: Vec::new(),
+                            script_sig: Script::new(),
+                        }],
+                        output: vec![TxOut {
+                            script_pubkey: target.script_pubkey(),
+                            value: *amount,
```

Comment:
Please add:

```
// TODO: deal with fee
```

At the meantime, please subtract a reasonable fee number (120sat per byte, and the size is constant, please measure).